job-server/src/spark.jobserver/JobInfoActor.scala

```diff
@@ -33,18 +33,19 @@ class JobInfoActor(jobDao: JobDAO, contextSupervisor: ActorRef) extends Instrume
 
   override def wrappedReceive: Receive = {
     case GetJobStatuses(limit) =>
-      sender ! jobDao.getJobInfos(limit.get)
+      val jobInfos = Await.result(jobDao.getJobInfos(limit.get), 60 seconds)
+      sender ! jobInfos
 
     case GetJobStatus(jobId) =>
       val jobInfo = Await.result(jobDao.getJobInfo(jobId), 60 seconds)
```

Comment:
Note I didn't test the above out .... I'm assuming Future has a collect method but it might not.  Either way the point is to not do any work after blocking in an actor .... actors should NEVER block.